wagtail/models/i18n.py

```diff
@@ -183,27 +183,33 @@ class TranslatableMixin(models.Model):
 
     class Meta:
         abstract = True
-        unique_together = [("translation_key", "locale")]
+        constraints = [
+            models.UniqueConstraint(
+                fields=["translation_key", "locale"],
+                name="unique_translation_key_locale_%(app_label)s_%(class)s",
+            )
+        ]
 
     @classmethod
     def check(cls, **kwargs):
         errors = super().check(**kwargs)
         is_translation_model = cls.get_translation_model() is cls
 
-        # Raise error if subclass has removed the unique_together constraint
+        unique_constraint_fields = ("translation_key", "locale")
+
+        # Raise error if subclass has removed constraints
         # No need to check this on multi-table-inheritance children though as it only needs to be applied to
         # the table that has the translation_key/locale fields
-        if (
-            is_translation_model
-            and ("translation_key", "locale") not in cls._meta.unique_together
-        ):
+        if is_translation_model and unique_constraint_fields in cls._meta.constraints:
```

Comment:
I think it's better to keep both the `unique_constraint` check and the new `UniqueConstraint` check. Also, perhaps you meant to write `unique_constraint_fields not in cls._meta.constraints` (the `not` is missing)? 

However, even with that, this new check will not work because `("translation_key", "locale") not in cls._meta.constraints` will always evaluate to `True`, since `cls._meta.constraints` will be a list of `models.BaseConstraint` instances, not tuples of field names.

You'll need something like

```suggestion
        has_unique_constraint = any(
            (
                isinstance(constraint, models.UniqueConstraint)
                and set(constraint.fields) == set(unique_constraint_fields)
            )
            for constraint in constraints
        )
        
        has_unique_together = unique_constraint_fields in cls._meta.unique_together
        
        if is_translation_model and not (has_unique_constraint or has_unique_together):
```

Reply:
Yeah, I think that was a mistake and I would correct that in the next PR. Thanks for pointing it out.